src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -92,10 +92,11 @@ class MessageBufferProcessor(
 		val batchSizeWithExtraGap = batchSize * 2
 		val messages = messageStore.loadLockedMessages(batchSizeWithExtraGap)
 		scala.util.Random.shuffle(producers).foreach {
-			case (exchange, producer) =>
-				val messagesToProducer = messages.filter(_.exchangeName == producer.exchange.name)
+			case (exchange: String, producer: AmqpProducerInterface) =>
```

Comment:
I think this change results in `isInstanceOf`calls which is not necessary. You could keep this line unchanged.